phoenix-core/src/main/java/org/apache/phoenix/coprocessor/MetaDataEndpointImpl.java

```diff
@@ -2082,7 +2082,7 @@ private long getViewIndexSequenceValue(PhoenixConnection connection, String tena
         long sequenceTimestamp = HConstants.LATEST_TIMESTAMP;
         try {
             connection.getQueryServices().createSequence(key.getTenantId(), key.getSchemaName(), key.getSequenceName(),
-                    Long.MIN_VALUE, 1, 1, Long.MIN_VALUE, Long.MAX_VALUE, false, sequenceTimestamp);
+                    Short.MIN_VALUE, 1, 1, Long.MIN_VALUE, Long.MAX_VALUE, false, sequenceTimestamp);
```

Comment:
This is the core change. In order to work with both short and long view index ids, we start with Short.MIN_VALUE, which can be encoded both as short and long.

The bulk of the rest are test fixes for tests that happen to verify the generate index id.